Address @casz review comments from EWM PR#68 (JCasC support)

BenjaminBeggs opened the pull request:
This PR will attempt to address all feedback from @casz for https://github.com/jenkinsci/external-workspace-manager-plugin/pull/68. Once the work is finished and the changes approved, this PR can be merged by a user with write access on this repository to have the commits contribute directly to https://github.com/jenkinsci/external-workspace-manager-plugin/pull/68.

**Review comments addressed**:
- Deprecated ConfigAsCodeTest class replaced with ConfiguredWithCode annotation to load .yaml configuration.
- JenkinsConfiguredWithCodeRule replaces JenkinsRule.
- Deleted unused "obsolete-configuration-as-code.yaml" file
- Export configuration unit tests have been rewritten
- Determined cause of commented out node configuration error (https://github.com/jenkinsci/external-workspace-manager-plugin/pull/68#discussion_r303040400 caused by indentation inconsistencies and `jnlp:` line)

@martinda @imaffe

Commit:
Address use of deprecated "@Test" annotation

Replaced with @ConfiguredWithCode.

Commit:
Update ConfigAsCodeTest.java

martinda reviewed (approved):
Looks good to me.

BenjaminBeggs commented:
If you feel comfortable with the changes you can merge them to have the commits contribute to https://github.com/jenkinsci/external-workspace-manager-plugin/pull/68 (JCasC support for EWM).

jetersen commented on src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java:
```diff
@@ -40,7 +41,7 @@
     @ClassRule public static JenkinsRule r = new JenkinsRule();
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
```
```suggestion
    @Test
```

jetersen commented on src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java:
```diff
@@ -59,7 +60,7 @@ public void shouldSupportConfigurationAsCodeExwsAllocateStep() throws Exception
 
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
```
```suggestion
    @Test
```

jetersen commented on src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java:
```diff
@@ -83,7 +84,7 @@ public void shouldSupportConfigurationAsCodeExwsStep() throws Exception {
     }
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
```
```suggestion
    @Test
```

jetersen commented on src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java:
```diff
@@ -97,7 +98,7 @@ public void shouldSupportConfigurationAsCodeMasterProperty() throws Exception {
     }
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
```
```suggestion
    @Test
```

jetersen commented on src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java:
```diff
@@ -108,7 +109,7 @@ public void shouldSupportConfigurationAsCodeAgentProperty() throws Exception {
     }
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
```
```suggestion
    @Test
```

jetersen commented on src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java:
```diff
@@ -97,7 +98,7 @@ public void shouldSupportConfigurationAsCodeMasterProperty() throws Exception {
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
     public void shouldSupportConfigurationAsCodeAgentProperty() throws Exception {
         URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
```
102-104 can be removed 😓 
```suggestion
```

jetersen commented on src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java:
```diff
@@ -83,7 +84,7 @@ public void shouldSupportConfigurationAsCodeExwsStep() throws Exception {
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
     public void shouldSupportConfigurationAsCodeMasterProperty() throws Exception {
         URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
```
88-90 can be removed 😓 
```suggestion
```

jetersen commented on src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java:
```diff
@@ -59,7 +60,7 @@ public void shouldSupportConfigurationAsCodeExwsAllocateStep() throws Exception
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
     public void shouldSupportConfigurationAsCodeExwsStep() throws Exception {
         URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
```
64-66 can be removed 😓 
```suggestion
```

jetersen commented:
This all wrong 😆 You basically removed the tests 😓

jetersen commented on src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java:
```diff
@@ -40,7 +41,7 @@
 
     @ClassRule public static JenkinsRule r = new JenkinsRule();
```
```suggestion
     @ClassRule
     @ConfiguredWithCode("configuration-as-code.yml")
     public static JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
```

jetersen commented on src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java:
```diff
@@ -108,7 +109,7 @@ public void shouldSupportConfigurationAsCodeAgentProperty() throws Exception {
 
-    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
     public void exportConfiguration() throws Exception {
```
This test needs to be completely rewritten

BenjaminBeggs replied:
I will rewrite this.

jetersen reviewed (changes requested):
See this for reference on an import/export test with a `@ClassRule`
https://github.com/jenkinsci/slack-plugin/blob/8acdef301a75b348f4d20794a4fd31c439c669b6/src/test/java/jenkins/plugins/slack/jcasc/ConfigurationAsCodeTest.java

BenjaminBeggs commented:
Haha! Thanks for the review, I’ll sort this out.

Commit:
Update src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

Co-Authored-By: Joseph Petersen <josephp90@gmail.com>

Commit:
Update src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

Co-Authored-By: Joseph Petersen <josephp90@gmail.com>

Commit:
Update src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

Co-Authored-By: Joseph Petersen <josephp90@gmail.com>

Commit:
Update src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

Co-Authored-By: Joseph Petersen <josephp90@gmail.com>

Commit:
Update src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

Co-Authored-By: Joseph Petersen <josephp90@gmail.com>

Commit:
Update ConfigAsCodeTest.java

Commit:
Update ConfigAsCodeTest.java

Commit:
Update ConfigAsCodeTest.java

Commit:
Update ConfigAsCodeTest.java

Commit:
Update ConfigAsCodeTest.java

Commit:
Update ConfigAsCodeTest.java

Commit:
Update ConfigAsCodeTest.java

Commit:
Delete obsolete-configuration-as-code.yaml

Unused.

Commit:
Commented out export test pending rewrite

BenjaminBeggs commented:
@casz I have implemented your suggestions. I think things are working properly now with the ConfiguredWithCode annotation. I am working on rewriting the unit test.

Commit:
Export unit tests rewritten

Commit:
Newline fix

Commit:
Update global-casc-templates-expected.yaml

BenjaminBeggs commented:
Export configuration unit tests have been fully rewritten @casz  @martinda

Commit:
Update configuration-as-code.yaml

Commit:
Update configuration-as-code.yaml

BenjaminBeggs commented:
I have corrected some configuration-as-code.yaml indentations and removed the line for
`jnlp:`
from the commented out nodes configuration code (for https://github.com/jenkinsci/external-workspace-manager-plugin/pull/68#discussion_r295124968). It was these things that were causing the test failures. It is unclear to me why the `jnlp:` line is causing a crash.

@martinda @casz
